### PR TITLE
IT VDT - Add Ubuntu22 support

### LINKED ISSUE #2836: Integration tests for Vulnerability Detector: Ubuntu 22 support
### Description

This issue is part of wazuh/wazuh#13162

As we currently perform integration tests for other Ubuntu versions, it is necessary that we include support for this new Ubuntu 22 (Jammy Jellyfish), which includes a new OVAL feed offered by Canonical – [OVAL Ubuntu Jammy](https://security-metadata.canonical.com/oval/com.ubuntu.jammy.cve.oval.xml.bz2).

The following tests should cover the new supported OS version:

- [Feed tests](https://github.com/wazuh/wazuh-qa/tree/master/tests/integration/test_vulnerability_detector/test_feeds): These tests have to cover the addition of the new feed for `Ubuntu 22`
- [Provider tests](https://github.com/wazuh/wazuh-qa/tree/master/tests/integration/test_vulnerability_detector/test_providers): Here we will test the settings for `Canonical` with OS version `Jammy`
- [Scan results tests](https://github.com/wazuh/wazuh-qa/tree/master/tests/integration/test_vulnerability_detector/test_scan_results): Here we have to add the tests that ensure `Ubuntu 22` agents are scanned properly

### PR DESCRIPTION
**Pull request by Deblintrake09**
|Related issue|
|---|
|Close #2836   |

## Description

For Wazuh v4.4.0 we want to add Ubuntu 2022 OS support for the Wazuh Vulnerability Detector module.

The following test was added to cover the new supported OS version:

- `Feed tests:` These tests have to cover the addition of the new feed for `Ubuntu 2022`
- `Provider tests`: Here we test the settings for `Ubuntu` with OS version `2022`
- `Scan results tests`: Here we  add the tests that ensure `Ubuntu 2022` agents are scanned properly

Modules Involved: 

- test_vulnerability_detector/test_feeds/test_download_feeds.py
- test_vulnerability_detector/test_feeds/test_duplicate_feeds.py
- test_vulnerability_detector/test_feeds/test_import_invalid_feed_type.py
- test_vulnerability_detector/test_feeds/test_validate_xml_feed_content.py
- test_vulnerability_detector/test_providers/test_enabled.py
- test_vulnerability_detector/test_providers/test_missing_os.py
- test_vulnerability_detector/test_providers/test_os.py
- test_vulnerability_detector/test_providers/test_update_interval.py
- test_vulnerability_detector/test_scan_results/test_scan_provider_and_nvd_vulnerabilities.py
- test_vulnerability_detector/test_scan_results/test_scan_provider_vulnerabilities.py
- test_vulnerability_detector/test_scan_results/test_scan_vulnerability_removal.py

## Tests Results
| Test Path | Os/Type | Execution Type | Results | Date | By |
|--|--|--|--|--|--|
|  test/integration/test_vulnerability_detector/| CentOS - Manager | Jenkins | [:large_blue_circle: ](https://ci.wazuh.info/view/Tests/job/Test_integration/27633/) | 2022/06/08 | @Deblintrake09 
|  test/integration/test_vulnerability_detector/| CentOS - Manager | Local |  [:green_circle:](https://github.com/wazuh/wazuh-qa/files/8781472/R1.zip) [:green_circle:](https://github.com/wazuh/wazuh-qa/files/8781473/R2.zip) [:green_circle:](https://github.com/wazuh/wazuh-qa/files/8781474/R3.zip) | 2022/05/26 | @Deblintrake09

## Tests

- [x] Proven that tests **pass** when they have to pass.
- [x] Proven that tests **fail** when they have to fail.

- [x] Python codebase satisfies PEP-8 style style guide. `pycodestyle --max-line-length=120 --show-source --show-pep8 file.py`.
- [x] Python codebase is documented following the Google Style for Python docstrings.

### CODE REVIEW

File: tests/integration/test_vulnerability_detector/test_feeds/data/test_cases/cases_download_feeds.yaml [hunk 1]
```diff
@@ -90,6 +90,18 @@
...
+- name: 'JAMMY'
+  description: 'Canonical provider'
+  configuration_parameters:
+    PROVIDER: 'canonical'
+    OS: 'jammy'
+  metadata:
+    provider_name: 'Ubuntu Jammy'
+    download_timeout: 120
+    provider_os: 'JAMMY'
+    update_treshold_weeks: 2
```
Comment (damarisg) [inline tests/integration/test_vulnerability_detector/test_feeds/data/test_cases/cases_download_feeds.yaml:102]: Did you check that it takes 2 weeks?

File: tests/integration/test_vulnerability_detector/data/feeds/canonical/custom_ubuntu22_oval_feed.xml [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+    <states>
+        <ind-def:family_state id="oval:com.ubuntu.jammy:ste:100" version="1" comment="The singleton family object.">
+            <ind-def:family>unix</ind-def:family>
+        </ind-def:family_state>
+
+        <ind-def:textfilecontent54_state id="oval:com.ubuntu.jammy:ste:101" version="1" comment="Ubuntu 22.04 LTS">
+            <ind-def:subexpression>jammy</ind-def:subexpression>
+        </ind-def:textfilecontent54_state>
+    </states>
+</oval_definitions>
```
[the remark this answers is not in the document]
Comment (Deblintrake09) [reply tests/integration/test_vulnerability_detector/data/feeds/canonical/custom_ubuntu22_oval_feed.xml]: Fixed

File: tests/integration/test_vulnerability_detector/test_feeds/data/test_cases/cases_download_feeds.yaml (hunk 1 above)
Comment (Deblintrake09) [reply tests/integration/test_vulnerability_detector/test_feeds/data/test_cases/cases_download_feeds.yaml:102]: Could not find any specific information on what it takes exactly.

**Review by damarisg — DISMISSED** [review]
| Test Path | Os/Type | Execution Type | Results | Date | By |
|--|--|--|--|--|--|
|  test/integration/test_vulnerability_detector/| CentOS - Manager | Jenkins | [🟢 ](https://ci.wazuh.info/view/Tests/job/Test_integration/26944/) | 2022/05/25/ |Seyla

**Comment by Deblintrake09** [comment]
| Test Path | Os/Type | Execution Type | Results | Date | By |
|--|--|--|--|--|--|
|  test/integration/test_vulnerability_detector/| CentOS - Manager | Jenkins | [:large_blue_circle: ](https://ci.wazuh.info/view/Tests/job/Test_integration/27078/) [:large_blue_circle: ](https://ci.wazuh.info/view/Tests/job/Test_integration/27079/) [:large_blue_circle: ](https://ci.wazuh.info/view/Tests/job/Test_integration/27080/) | 2022/05/31 | @Deblintrake

File: tests/integration/test_vulnerability_detector/test_feeds/data/test_cases/cases_validate_xml_feed_content.yaml [hunk 3]
```diff
@@ -53,7 +64,7 @@
     decompressed_file: '/tmp/focal.xml'
     url: 'https://people.canonical.com/~ubuntu-security/oval/com.ubuntu.focal.cve.oval.xml.bz2'
 
-- name: 'Canonical'
+- name: 'Canonical Bionic'
```
[the remark this answers is not in the document]
Comment (Deblintrake09) [reply tests/integration/test_vulnerability_detector/test_feeds/data/test_cases/cases_validate_xml_feed_content.yaml:67]: Done

File: tests/integration/test_vulnerability_detector/test_providers/data/test_cases/cases_update_interval.yaml [hunk 4]
```diff
@@ -28,6 +28,46 @@
...
+  description: 'Test update interval 5s Ubuntu Trusty'
+  configuration_parameters:
+    PROVIDER: 'canonical'
+    OS: 'trusty'
+    UPDATE_INTERVAL: '5s'
+  metadata:
+    provider_name: 'Ubuntu Trusty'
+    update_interval: '5s'
+
+- name: 'Ubuntu Jammy'
```
[the remark this answers is not in the document]
Comment (Deblintrake09) [reply tests/integration/test_vulnerability_detector/test_providers/data/test_cases/cases_update_interval.yaml]: Fixed in https://github.com/wazuh/wazuh-qa/pull/2920/commits/afb87adf6c8a076d81fbc90dc7c32fbd97d207d6

File: deps/wazuh_testing/wazuh_testing/mocking/__init__.py [hunk 5]
```diff
@@ -103,7 +103,9 @@
     'BUSTER': {'os_name': 'Debian GNU/Linux', 'os_major': '10', 'os_minor': '0', 'os_platform': 'debian',
                'name': 'debian10'},
     'STRETCH': {'os_name': 'Debian GNU/Linux', 'os_major': '9', 'os_minor': '0', 'os_platform': 'debian',
-                'name': 'debian9'}
+                'name': 'debian9'},
+    'JAMMY': {'os_name': 'Ubuntu', 'os_major': '22', 'os_minor': '04', 'os_platform': 'ubuntu',
```
[the remark this answers is not in the document]
Comment (Deblintrake09) [reply deps/wazuh_testing/wazuh_testing/mocking/__init__.py]: Fixed in https://github.com/wazuh/wazuh-qa/pull/2920/commits/afb87adf6c8a076d81fbc90dc7c32fbd97d207d6

File: tests/integration/test_vulnerability_detector/test_scan_results/test_scan_provider_and_nvd_vulnerabilities.py [hunk 6]
```diff
@@ -135,6 +136,9 @@ def test_scan_provider_and_nvd_vulnerabilities(configuration, metadata, agent_sy
         - f"The '{package}' package .* from agent '{agent_id}' is vulnerable to '{cve}'"
         - f".*"agent":."id":"{agent_id}".*{cve} affects {package}"
     '''
+    if agent_system == 'JAMMY':
```
[the remark this answers is not in the document]
Comment (Deblintrake09) [reply tests/integration/test_vulnerability_detector/test_scan_results/test_scan_provider_and_nvd_vulnerabilities.py:143]: We need to generate an Issue to track this for when it is added, so we can remove the Skip

File: tests/integration/test_vulnerability_detector/test_feeds/data/test_cases/cases_duplicate_feeds.yaml [hunk 7]
```diff
@@ -20,7 +20,27 @@
     provider_name: 'Debian Buster'
     provider_json_name: ''
 
-- name: 'Ubuntu'
+- name: 'Ubuntu Trusty'
```
[the remark this answers is not in the document]
Comment (Deblintrake09) [reply tests/integration/test_vulnerability_detector/test_feeds/data/test_cases/cases_duplicate_feeds.yaml:23]: This was added to prodive full coverage on the different System versions.

**Comment by jmv74211** [comment]
Closed since these changes have been merged with the `master` <-- `4.4`

**Pull request closed without merging**